zaas-service/src/main/java/org/zowe/apiml/zaas/security/service/token/OIDCTokenProvider.java

```diff
@@ -98,9 +97,11 @@ void fetchJWKSet() {
         try {
             publicKeys.clear();
             jwkSet = null;
-            Resource resource = resourceRetriever.retrieveResource(new URL(jwksUri));
-            jwkSet = JWKSet.parse(resource.getContent());
-            publicKeys.putAll(processKeys(jwkSet));
+            for (String url : jwksUri) {
+                Resource resource = resourceRetriever.retrieveResource(new URL(url));
+                jwkSet = JWKSet.parse(resource.getContent());
+                publicKeys.putAll(processKeys(jwkSet));
+            }
         } catch (IOException | ParseException | IllegalStateException e) {
             log.error("Error processing response from URI {} message: {}", jwksUri, e.getMessage());
```

Comment:
should we maybe move the try catch inside the for and log just the failing jwk URI?